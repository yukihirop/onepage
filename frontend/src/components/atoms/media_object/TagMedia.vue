<template lang="pug">
.a-tag-media
  .a-tag-media-rank
    p(style="margin-top: 10px;") {{ rank }}
  .a-tag-media-image(style="width: 64px;")
    img(:src='tagImageSrc', style="width: 32px;margin-top: 10px;")
  .a-tag-media-name
    b
      | {{ tagname }}
  .a-tag-media-post
    .a-tag-media-post-posts
      b {{ posts }}
    .a-tag-media-post-label
      small posts
</template>

<script>
export default {
  props: {
    rank: {
      type: Number,
      required: true
    },
    tagImageSrc: {
      type: String,
      required: true
    },
    tagname: {
      type: String,
      required: true
    },
    posts: {
      type: Number,
      required: true
    }
  },
  // https://kuroeveryday.blogspot.jp/2017/10/img-src-data-binding-vuejs.html
  data () {
    return {
      src: require(this.tagImageSrc)
    }
  }
}
</script>

<style scoped lang="stylus">
.a-tag-media
  display: flex
  flex-direction: row
  align-items: center
  justify-content: space-between
  width: 100%
  
  &-rank
    width: 20px
  
  &-name
    width:100px
    text-align: left
    padding-top: 10px
    padding-bottom: 10px
  
  &-post
    display: flex
    flex-direction: column
    align-items: center
    margin-left: auto
    
    &-posts
      width: 50px
      text-align: left
    &-label
      width: 50px
      text-align:left
</style>