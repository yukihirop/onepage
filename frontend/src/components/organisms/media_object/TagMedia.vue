<template lang="pug">
nav.panel
  .panel-heading
    .title.is-6 タグ・ランキング
  .panel-tabs
    a.is-active 週間
    a 月間
    a 全て
  tag-media(rank=1, :tag-image-src='require("../../../assets/home/tag/Python.jpg")', tagname='Python',     posts=99999)
  tag-media(rank=2, :tag-image-src='require("../../../assets/home/tag/JavaScript.jpg")', tagname='JavaScript', posts=9999)
  tag-media(rank=3, :tag-image-src='require("../../../assets/home/tag/Ruby.jpg")', tagname='Ruby',       posts=8888)
  tag-media(rank=4, :tag-image-src='require("../../../assets/home/tag/AWS.jpg")', tagname='AWS',        posts=7777)
  tag-media(rank=5, :tag-image-src='require("../../../assets/home/tag/Android.jpg")', tagname='Android',    posts=6666)
  tag-media(rank=6, :tag-image-src='require("../../../assets/home/tag/Docker.png")', tagname='Docker',     posts=5555)
  tag-media(rank=7, :tag-image-src='require("../../../assets/home/tag/Swift.png")', tagname='Swift',      posts=4444)
  tag-media(rank=8, :tag-image-src='require("../../../assets/home/tag/iOS.png")', tagname='iOS',        posts=333)
  tag-media(rank=9, :tag-image-src='require("../../../assets/home/tag/Rails.jpg")', tagname='Rails',      posts=22)
  tag-media(rank=10,:tag-image-src='require("../../../assets/home/tag/PHP.jpg")', tagname='PHP',       posts=1)
</template>

<script>
import TagMedia from '../../atoms/media_object/TagMedia.vue'

export default {
  components: {
    TagMedia
  }
}
</script>

<style scoped lang="stylus">
$custome_white = #F8F8F8
$theme_color = #0081C1

nav.panel
  a
    color: $theme_color
    .is-active
      color: dark

.panel-heading
  margin-bottom: 0px
  background-color: $theme_color
  .title
    color: white
.panel-tabs
  padding-bottom: 0px
  margin-bottom: 0px
  background-color: $custome_white
</style>