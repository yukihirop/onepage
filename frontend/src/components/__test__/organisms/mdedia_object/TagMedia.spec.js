import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import TagMedia from '@/components/organisms/media_object/TagMedia.vue'
import factoryTags from '@/components/__test__/factories/tags'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('TagMedia.vue', () => {
  it('レンダリングのテスト', () => {
    // shallowは子コンポーネントまで入り込まない。
    // <!---->のように置き換えてしまう
    // そのため？子との共有プロパティーである?
    // tagImageSrcに値が渡らない
    // tagImageSrcはrequiredなので実行時にエラーがでるので
    // falseにして上書きしている
    // 下記エラーの回避策
    // console.error node_modules/vue/dist/vue.runtime.common.js:576
    // [Vue warn]: Missing required prop: "tagImageSrc"
    const wrapper = shallow(TagMedia, {
      localVue,
      props: {
        tagImageSrc: {
          type: String,
          required: false
        }
      }
    })
    wrapper.setData(factoryTags.data())

    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
