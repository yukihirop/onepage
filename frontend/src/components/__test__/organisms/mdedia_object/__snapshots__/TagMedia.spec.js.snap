// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagMedia.vue レンダリングのテスト 1`] = `
<nav class="panel">
    <div class="panel-heading">
        <div class="title is-6">タグ・ランキング</div>
    </div>
    <div class="panel-tabs">
        <a class="is-active">週間</a>
        <a>月間</a>
        <a>全て</a>
    </div>
    <div class="panel-block">
        <!---->
    </div>
    <div class="panel-block">
        <!---->
    </div>
</nav>
`;
